README.md

```diff
@@ -1,9 +1,111 @@
 
+## Overview
+Replication is the process of creating a copy of a subset of data and storing it on another DDF or HDFS based system. Data can be pulled from a remote DDF and saved to another DDF or HDFS system. Metacards produced by replication are marked with a "Replication Origins" attribute and a tag of "replicated". Replication will automatically start transferring data once an admin creates a replication configuration.
+
+## Installing Replication
+Replication is not installed by default with a standard installation. There are several installation options available.
+
+#### Installing Via Admin Console
+An administrator can install the replication feature by following these steps:
+* Navigate to the Admin Console.
+* Select the System tab.
+* Select the Features tab.
+* Locate the replication feature.
+* Select the Install arrow for the replication feature.
+* The status will change from Uninstalled to Installed.
+
+#### Installing Via Kar Feature File
+An administrator can install the replication feature by following the following steps:
+* Locate and download the desired replication artifact kar file.
+* On a running and configured DDF instance, place the kar file in the ${DDF_HOME}/deploy directory.
+* Confirm the feature is running in the Admin Console or the Karaf command line interface.
+
+## Configuring Replication Nodes
+This version of Replication supports nodes created for DDF based applications as well as webHDFS based systems.
+
+#### Configuring DDF Nodes
+To configure replication to/from a DDF node, a node must first be created. This can be done in a few steps:
+* Navigate to https://<host>:<port>/admin/replication/#/nodes and click the + 
+* On the Create new Node screen, fill out the following fields:
+  * Name: The display name to give the node. 
+  * Hostname: The hostname of the node.
+  * Port: The port where the service can be found. This is `8993` by default in DDF.
+  * Root Context: The path under which replication services can be found. This is `services` by default in DDF.
+  
+#### Configuring WebHDFS Nodes
+To configure replication to/from a HDFS node, a node must first be created. This can be done in a few steps:
+* Navigate to https://<host>:<port>/admin/replication/#/nodes and click the + 
+* On the Create new Node screen, fill out the following fields:
+  * Name: The display name to give the node. 
+  * Hostname: The hostname of the node.
+  * Port: The port where the service can be found. This is `9870` by default for HDFS Namenodes.
+  * Root Context: This is the full path after the port for any webHDFS URL.
+
+* localhost webHDFS example configuration : 
+```
+To write to directory /user/tonystark on a local webHDFS instance
+- The webHDFS URL for that directory might look like: http://localhost:9870/webhdfs/v1/user/tonystark
+- The hostname would be: localhost
+- The port would be: 9870
+- The root context would be /webhdfs/v1/user/tonystark/
```

Comment:
Is this leading `/` necessary?

Reply:
Isn't it? 

`String baseUrl = "http://" + url.getHost() + ":" + url.getPort() + url.getPath();`

We're not using exactly what is passed in. We do add the trailing slash to the path if it is missing, but I did not add prepending a slash to the path if it is missing.

Reply:
I suppose it depends on how the URL is constructed from the replication configuration which gets passed into the factory `create` method.  Within that method it is broken apart and then rebuilt.  In addition, the DDF "default" for this field does not have a leading `/` before services.  I haven't tried it today, but it's worth confirming.

Reply:
Good point. I will verify this is a requirement for the path.

Reply:
I tried it without the leading slash and it did work. The UI does show the complete URL without me typing in the leading slash: `https://localhost:9870/webhdfs/v1/user/aaronilovici`

And the dib.log shows the full URL of the file being replicated including the leading slash which I did not enter in the node setup:
`The complete file URL is: http://localhost:9870/webhdfs/v1/user/aaronilovici/4ad5e49e40944713945abc6fa8775ed4_1591107095101.txt`

I can remove the leading slash from the README examples.